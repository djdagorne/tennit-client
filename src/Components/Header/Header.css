header {
  margin: 45px;
}


nav {
  position: fixed;
  height: 45px;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
}

.logo{
  float:left;
  font-family: 'Lato', sans-serif;
  text-align:center;
  margin-top: 6px;
  margin-left: 20px;

  font-size: 2.3em;
  font-weight: 900;
  color: #F0F0F2;
  text-decoration: none;
  letter-spacing: -2px;
  text-shadow: 0px 3px 10px rgba(119, 119, 119, 0.493);
  line-height: 1.5em;
}
.logo:hover{
  text-shadow: 0px 3px 10px rgba(201, 201, 201, 0.3);
}

.log-button, .account-button{
  font-family: 'Raleway', sans-serif;
  -webkit-appearance: none;
  background-color: transparent;
  border-style: none;
  position: absolute;
  text-shadow: 0px 3px 10px rgba(119, 119, 119, 0.493);
  letter-spacing: -1px;
  line-height: 1.5em;
  top: 20px;
  
  padding: 0px 6px;
  border-radius: 2px;
  float:right;
  color: #F0F0F2;
  font-size: 1.3em;
  text-decoration: none;
}
.log-button{
  right: 20px;
}
.log-button:hover{
  cursor: pointer;
}
.account-button{
  right: 85px;
}
.account-button:hover{
  cursor: pointer;
} 