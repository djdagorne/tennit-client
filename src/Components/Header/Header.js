import React, {Component} from 'react';
import { Link } from 'react-router-dom'
import './Header.css';
import LogIn from '../Pages/LogIn/LogIn.js'
import EditAccount from '../Pages/EditAccount/EditAccount'
import CreateAccount from '../Pages/CreateAccount/CreateAccount'
import TennitContext from '../../TennitContext'
import TokenService from '../../Services/TokenService';

class Header extends Component {
    static contextType = TennitContext
    constructor(props){
        super(props);
        this.state = {
        }
    };

    renderLogInLink = () =>{
        return(
            <nav className="nav-not-logged-in">
                <Link to="/" className="logo text-shadow">tenn.it</Link>
                <button className="log-button text-shadow" onClick={e=>this.context.togglePopup('login')}>log in (demo)</button>
                {/* <button className="account-button text-shadow" onClick={e=>this.context.togglePopup('create')}>sign up</button> */}
           </nav>
        )
    }

    renderLogOutLink = () =>{
        return(
            <nav className="nav-logged-in">
                <Link to="/home" className="logo text-shadow">tenn.it</Link>
                <Link to="/"><button  onClick={this.context.toggleLogIn} className="log-button text-shadow">log out</button></Link>
                <button onClick={e=>this.context.togglePopup('edit')} className="account-button text-shadow">edit profile</button>
            </nav>
        )
    }


    render(){
        return (
            <>
                {TokenService.hasAuthToken() ? 
                    this.renderLogOutLink() :
                    this.renderLogInLink()
                }

                {this.context.showLogInPopup ?
                        <LogIn /> :
                        null
                    }

                {this.context.showCreatePopup ? 
                    <CreateAccount /> :
                    null
                }

                {this.context.showEditPopup ? 
                    <EditAccount /> :
                    null
                }
            </>
        );
    }
}

export default Header;