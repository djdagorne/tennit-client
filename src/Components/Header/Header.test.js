import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './Header'
import '../../setupTest'

it('renders without crashing', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
})
