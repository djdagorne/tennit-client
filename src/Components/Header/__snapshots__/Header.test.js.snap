// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <nav
    className="nav-not-logged-in"
  >
    <a
      className="logo text-shadow"
      href="/"
    >
      tenn.it
    </a>
    <button
      className="log-button text-shadow"
      onClick={[Function]}
    >
      log in (demo)
    </button>
    <button
      className="account-button text-shadow"
      onClick={[Function]}
    >
      sign up
    </button>
  </nav>
</Fragment>
`;
